GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs

```diff
@@ -315,10 +332,20 @@ public void Respawn()
             RespawnTimer = -1;
         }
 
-	    public void Recall(ObjAIBase owner)
+	    public void Recall(ObjAIBase owner,float timer)
         {
             var spawnPos = GetRespawnPosition();
-            _game.PacketNotifier.NotifyTeleport(owner, spawnPos.X, spawnPos.Y);
+            isRecalling = true;
+            visualBuff = ApiFunctionManager.AddBuffHUDVisual("Recall", timer, 1, this);
+            addParticle = ApiFunctionManager.AddParticleTarget(this, "TeleportHome.troy", this);
+
+            ApiFunctionManager.CreateTimer(timer, () =>
+            {
+                if (canRecall && !wasInterrupted)
```

Comment:
also it seems like you want the next 2 lines in that `if`. you can't do that without using `{ }`.